AsyncDisplayKit/ASDisplayNode+Subclasses.h

```diff
@@ -238,36 +238,58 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState ASDISPLAYNODE_REQUIRES_SUPER;
 
 /**
- * @abstract Called whenever the visiblity of the node changed.
+ * @abstract Called whenever the node becomes visible.
  *
  * @discussion Subclasses may use this to monitor when they become visible.
+ *
+ * @note This method can be called from any thread and should therefore be thread safe.
```

Comment:
@garrettmoon this one isn't right — we should make a strong guarantee that the visible state methods are called on the main thread.  It's the only way to guarantee their timing is relevant (e.g., actually called before it is visible onscreen), and work performed here is very often related to animations or other main thread things